adagrams/game.py

```diff
@@ -1,11 +1,104 @@
+import random
+
 def draw_letters():
-    pass
+    letter_pool = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    final_letter_choice = []
+    for number in range(0,10):
+        chosen_letter = random.choice(list(letter_pool.keys()))
+        if (letter_pool[chosen_letter] == 1):
+            letter_pool.pop(chosen_letter)
+        else:
+            letter_pool[chosen_letter] -= 1
+        final_letter_choice.append(chosen_letter)
+        
+    return final_letter_choice
```

Comment:
This way of choosing letters randomly doesn't quite meet the requirements. Notice that your solution is just as likely to draw a `Z` as it is to draw an `E`. The requirements say, "Since there are 12 `E`s but only 1 `Z`, it should be 12 times as likely for the user to draw an `E` as a `Z`". It turns out, our tests don’t test for that. (That test is quite hard to write!) Think about how you might modify your code to fulfill this requirement.

Regarding `random.choice`: when bringing in functions from imported modules (that we haven’t covered in the curriculum), please consider including a comment about how you understand the function to work, and how you researched it. Now that we’ve discussed Big O, it’s also useful to think about what the time/space complexity of the function might be. Thinking about how we can implement this functionality ourselves can help us gain insight into what that might be.

Remember that not every language has the same built-in functions, so we should be able to implement this same functionality ourselves. In the projects, we don’t ask for any features that we don’t think you should be able to write yourself. For drawing a random hand, the only external function that would be “required” is a way to pick a random number (such as randint). At this stage in our coding journeys, it’s often more valuable to re-implement things by hand to practice thinking about how to break down and approach the problem than to use a single library function to solve the problem for us.